lib/lighthouse/benefits_documents/service.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require 'common/client/base'
+require 'lighthouse/benefits_documents/configuration'
+require 'lighthouse/benefits_documents/service_exception'
+require 'lighthouse/service_exception'
+
+module BenefitsDocuments
+  class Service < Common::Client::Base
+    configuration BenefitsDocuments::Configuration
+    STATSD_KEY_PREFIX = 'api.benefits_documents'
+    STATSD_UPLOAD_LATENCY = 'lighthouse.api.benefits.documents.latency'
+
+    def initialize(icn)
+      @icn = icn
+      raise ArgumentError, 'no ICN passed in for LH API request.' if icn.blank?
+
+      super()
+    end
+
+    def queue_document_upload(params, lighthouse_client_id = nil)
+      start_timer = Time.zone.now
+      claim_id = params[:claimId]
+      raise Common::Exceptions::ArgumentError, claim_id unless claim_id
+
+      jid = submit_document(params[:file], claim_id, params[:trackedItemId], params[:documentType], params[:password],
+                            lighthouse_client_id)
+      StatsD.measure(STATSD_UPLOAD_LATENCY, Time.zone.now - start_timer, tags: ['is_multifile:false'])
+      cleanup_after_upload
+      jid
+    end
+
+    def queue_multi_image_upload_document(params, lighthouse_client_id = nil)
+      start_timer = Time.zone.now
+      claim_id = params[:claimId]
+      raise Common::Exceptions::ArgumentError, id unless claim
```

Comment:
On L24, you are raising like this:
```ruby
raise Common::Exceptions::ArgumentError, claim_id unless claim_id
```
Is there a reason why you use `id` here instead of `claim_id`?

Reply:
+1

Reply:
Oh shoot, you're right. I forgot to sync up the errors they throw after I updated `queue_document_upload`. Updated